ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift

```diff
@@ -89,6 +91,107 @@ public class SentryManager {
+    log.logln(event.message?.formatted, level: level)
+  }
+
+  /**
+   * Captures a Sentry event and copies its message to the engine's logging mechanism.
+   * If the logging level is not specified, the Sentry event's log-level will be used as a default.
+   *
+   * Will safely bypass the Sentry component if not activated by the app, only logging the
+   * message in such scenarios.
+   */
+  public static func captureAndLog(_ error: Error, message: String? = nil, sentryLevel: Sentry.SentryLevel = .error, logLevel: XCGLogger.Level? = nil) {
+    let event = Sentry.Event(error: error)
+    event.level = sentryLevel
+    if let message = message {
+      event.message = SentryMessage(formatted: message)
+    }
+
+    self.captureAndLog(event, logLevel: logLevel)
+  }
+
+  /**
+   * Constructs a SentryEvent around a message and also passes it to the engine's logging mechanism.
+   * If the logging level is not specified, it will default to .error.
+   *
+   * Will safely bypass the Sentry component if not activated by the app, only logging the
+   * message in such scenarios.
+   */
+  public static func captureAndLog(_ message: String, sentryLevel: Sentry.SentryLevel = .error, logLevel: XCGLogger.Level? = nil) {
+    let event = Sentry.Event(level: sentryLevel)
+    event.message = SentryMessage(formatted: message)
+
+    self.captureAndLog(event, logLevel: logLevel)
+  }
+
+  /**
+   * Adds a Sentry breadcrumb and copies its message to the engine's logging mechanism.
+   * If the logging level is not specified, the Sentry event's log-level will be used as a default.
+   *
+   * Will safely bypass the Sentry component if not activated by the app, only logging the
+   * message in such scenarios.
+   */
+  public static func breadcrumbAndLog(crumb: Sentry.Breadcrumb, logLevel: XCGLogger.Level? = nil) {
+    // Guarded in case a library consumer decides against initializing Sentry.
+    if _started {
+      SentrySDK.addBreadcrumb(crumb: crumb)
+    }
+
+    let level = logLevel ?? mapLoggingLevel(crumb.level)
+    log.logln(crumb.message, level: level)
+  }
+
+  /**
+   * Adds a Sentry breadcrumb and copies its message to the engine's logging mechanism.
+   * If the logging level is not specified, the Sentry event's log-level will be used as a default.
+   *
+   * Will safely bypass the Sentry component if not activated by the app, only logging the
+   * message in such scenarios.
+   */
+  public static func breadcrumbAndLog(_ message: String, category: String? = nil, sentryLevel: Sentry.SentryLevel = .info, logLevel: XCGLogger.Level? = nil) {
+    let crumb = Sentry.Breadcrumb()
+    crumb.level = sentryLevel
+    if let category = category {
+      crumb.category = category
+    }
+
+    self.breadcrumbAndLog(crumb: crumb, logLevel: logLevel)
+  }
+
   public static func forceError() {
     SentrySDK.addBreadcrumb(crumb: Sentry.Breadcrumb(level: .info, category: "Deliberate testing error"))
```

Comment:
Should these also be in
```
if _started {
}
```

Granted, this is outside the scope of the PR

Reply:
The functions they call already perform that filtering.  Doing so here would actually prevent the local 'log' statements from executing, and we don't want those blocked.

It's possible a library user may not wish to utilize Sentry.  In such cases, we should still make sure the on-device logs receive any error message sent through this helper.